packages/packages/documents/src/store/__tests__/sync.test.ts

```diff
@@ -0,0 +1,282 @@
+		const currentDocument = storeState.documents.documents[ currentDocumentId ];
+
+		expect( currentDocument.isSaving ).toBe( true );
+	} );
+
+	it( 'should sync saving state of a document on save', () => {
+		// Arrange.
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				makeMockDocument(),
+			] ),
+		};
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		const getCurrentDocument = () => {
+			const storeState = store.getState();
+			const { currentDocumentId } = storeState.documents;
+
+			return storeState.documents.documents[ currentDocumentId ];
+		};
+
+		// Act.
+		dispatchCommandBefore( 'document/save/save' );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( true );
+		expect( getCurrentDocument().isSavingDraft ).toBe( false );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save' );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( false );
+		expect( getCurrentDocument().isSavingDraft ).toBe( false );
+	} );
+
+	it( 'should sync draft saving state of a document on save', () => {
+		// Arrange.
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				makeMockDocument(),
+			] ),
+		};
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		const getCurrentDocument = () => {
+			const storeState = store.getState();
+			const { currentDocumentId } = storeState.documents;
+
+			return storeState.documents.documents[ currentDocumentId ];
+		};
+
+		// Act.
+		dispatchCommandBefore( 'document/save/save', {
+			status: 'autosave',
+		} );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( false );
+		expect( getCurrentDocument().isSavingDraft ).toBe( true );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save', {
+			status: 'autosave',
+		} );
+
+		// Assert.
+		expect( getCurrentDocument().isSaving ).toBe( false );
+		expect( getCurrentDocument().isSavingDraft ).toBe( false );
+	} );
+
+	it( 'should sync modified state of a document on V1 load', () => {
+		// Arrange.
+		const mockDocument = makeMockDocument();
+
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [ {
+				...mockDocument,
+				config: {
+					...mockDocument.config,
+					revisions: {
+						current_id: 2,
+					},
+				},
+			} ] ),
+		};
+
+		// Act.
+		dispatchV1ReadyEvent();
+
+		// Assert.
+		const storeState = store.getState();
+		const { currentDocumentId } = storeState.documents;
+		const currentDocument = storeState.documents.documents[ currentDocumentId ];
+
+		expect( currentDocument.isModified ).toBe( true );
+	} );
+
+	it( 'should sync modified state of a document on document change', () => {
+		// Arrange.
+		const mockDocument = makeMockDocument();
+
+		extendedWindow.elementor = {
+			documents: makeDocumentsManager( [
+				mockDocument,
+			] ),
+		};
+
+		// Populate the documents state.
+		dispatchV1ReadyEvent();
+
+		// Mock a change.
+		extendedWindow.elementor = {
```

Comment:
Maybe also create a dedicated function, it's really making me :(

Reply:
Done